components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java

```diff
@@ -1052,4 +1059,23 @@ private void updateMappedGroupForAttribute(ProvisioningEntity provisioningEntity
             ProvisioningUtil.setClaimValue(groupAttributeName, provisioningEntity.getAttributes(), mappedGroups);
         }
     }
+
+    /**
+     * Check if user roles has common elements with the provisioning role list
+     *
+     * @param userRoles
+     * @param provisioningRoles
+     */
+    private boolean userHasProvisioningRoles(List<String> userRoles, List<String> provisioningRoles, String userName) {
+
+        if (CollectionUtils.isNotEmpty(userRoles) && CollectionUtils.isNotEmpty(provisioningRoles) && !ListUtils
+                .intersection(userRoles, Arrays.asList(provisioningRoles)).isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("User with userName " + userName + " has provisioning role(s) assigned.");
```

Comment:
General pattern for log is like this(please note colon)
  log.debug("User with userName : " + userName + " has provisioning role(s) assigned.");

Reason, lets assume a user having a name "not found.". Can you imagine what is the log line printed.

Reply:
Also 
ListUtils.intersection() still creates another which is unnecessary for the purpose